learn-data-structures-by-building-a-shopping-cart/index17a.js

```diff
@@ -0,0 +1,65 @@
+const products = [
+  {
+    id: 0,
+    name: "Vanilla buttercream cupcake",
+    price: 2.99
+  },
+  {
+    id: 1,
+    name: "French Macaroon",
+    price: 3.99
+  },
+  {
+    id: 2,
+    name: "Fruit sprinkles cupcake",
+    price: 3.99
+  },
+  {
+    id: 3,
+    name: "Pink flower cupcake",
+    price: 5.99
+  }
+];
+
+class ShoppingCart {
+  constructor() {
+    this.items = [];
+    this.discountPercentage = 30;
+  }
+
+  addItem(id, products) {
+    const item = products.find(item => item.id === id);
+    this.items.push(item);
+  }
+
+  getItems() {
+    return this.items;
+  }
+  
+  getCount() {
+    return this.items.length;
+  }
+  
+  removeItem(id) {
+    const index = this.items.indexOf(id);
+    this.items.splice(index, 1);
+  }
+  
+  clearCart() {
+    this.items = [];
+  }
+  
+  applyDiscount(amount) {
+    return (this.discountPercentage / 100 * amount).toFixed(2);
+  }
+}
+
+/*
+toFixed method inadvertently converts a number into a string.
```

Comment:
One thing to keep in mind is that the `toFixed()` method converts numbers into strings.
Let's log a discounted price to the console and check the output:
```
const shoppingCart = new ShoppingCart();
console.log(shoppingCart.applyDiscount(2.99));
console.log(typeof shoppingCart.applyDiscount(2.99));
```